arangod/GeneralServer/RestHandler.cpp

```diff
@@ -100,38 +104,186 @@ void RestHandler::setStatistics(RequestStatistics* stat) {
   }
 }
 
+void RestHandler::forwardRequest() {
+  // TODO refactor into a more general/customizable method
+  //
+  // The below is mostly copied and only lightly modified from
+  // RestReplicationHandler::handleTrampolineCoordinator; however, that method
+  // needs some more specific checks regarding headers and param values, so we
+  // can't just reuse this method there. Maybe we just need to implement some
+  // virtual methods to handle param/header filtering?
+
+  // TODO verify that vst -> http -> vst conversion works correctly
+
+  // TODO verify that async requests work correctly
+
+  // TODO verify that user-JWT works correctly
+
+  uint32_t shortId = forwardingTarget();
+  std::string serverId =
+      ClusterInfo::instance()->getCoordinatorByShortID(shortId);
+  LOG_TOPIC(DEBUG, Logger::REQUESTS) << "forwarding request " << _request->messageId() << " to " << serverId;
```

Comment:
No error checking here if `serverId` is an empty string (not found in Agency).
I think in this case we must abort forwarding.